src/renderer/object/instanced_model.rs

```diff
@@ -49,53 +52,173 @@ impl<M: ForwardMaterial> InstancedModel<M> {
             instance_buffer1: InstanceBuffer::new(context)?,
             instance_buffer2: InstanceBuffer::new(context)?,
             instance_buffer3: InstanceBuffer::new(context)?,
+            instance_buffer4: InstanceBuffer::new(context)?,
             aabb,
             aabb_local: aabb.clone(),
             transformation: Mat4::identity(),
             transformations: transformations.to_vec(),
+            subtexture: None,
+            subtextures: None,
             material,
         };
-        model.update_transformations(transformations);
-        model.update_aabb();
+        model.set_transformations(transformations);
         Ok(model)
     }
 
     ///
-    /// Updates the transformations applied to each model instance before they are rendered.
-    /// The model is rendered in as many instances as there are transformation matrices.
+    /// Use [set_transformations()](InstancedModel::set_transformations()) instead.
     ///
+    #[deprecated]
     pub fn update_transformations(&mut self, transformations: &[Mat4]) {
+        self.set_transformations(&transformations);
+    }
+
+    ///
+    /// Store the transformations applied to each model instance before they are rendered.
+    /// The model is rendered in as many instances as there are transformation matrices.
+    ///
+    pub fn set_transformations(&mut self, transformations: &[Mat4]) {
```

Comment:
`set_instances` function that takes a list of instance definitions (transformation + texture transform)?